### ISSUE TITLE
Incorrect duration for mp4 audio from URL

### ISSUE THREAD

**Issue by yhcharles** [ryanheise/just_audio#694]
**Which API doesn't behave as documented, and how does it misbehave?**
When try to play an audio in mp4 format from URL, the duration in player shows as double time of actually value.

**Minimal reproduction project**
example:
https://github.com/yhcharles/just_audio/blob/double-duration/just_audio_background/example/lib/main.dart

**To Reproduce (i.e. user steps, not code)**
1. clone the given git repo, and checkout the `double-duration` branch
2. run the example in just_audio_background/example/lib/main.dart
3. click the play button to add an audio, which is pointing to the youtube video here: https://music.youtube.com/watch?v=1vrEljMfXYo
4. check the duration is 6min+
5. copy the URL for the mp4 media, and open it in browser, we can see the actual duration is 3min+

**Error messages**
The duration of this audio is shown as 06:36, but the actually value is 03:19, as we can see in the youtube page

**Expected behavior**
We should expect the correct duration showed in player, which is 03:19

**Screenshots**
![2](https://user-images.githubusercontent.com/2959773/160225199-903bd064-8a0b-4a5c-90ba-d90604fc8213.gif)

**Smartphone (please complete the following information):**
 - Device: iPhone13 simulator
 - OS: iOS 15.4

**Flutter SDK version**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, 2.10.3-0.0.pre.1, on macOS 12.3 21E230 darwin-arm, locale en-US)
[!] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.65.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.
```

**Additional context**

**Comment by ryanheise** [comment]
There might be too many layers happening here to understand what's going on. Can you make a minimal reproduction project reset involving YouTube and just a direct URL that I can examine?

**Comment by yhcharles** [comment]
@ryanheise simplified the example, and updated the screen record to show the problem. Thanks!

**Comment by ryanheise** [comment]
To investigate this, I need an actual mp4 file that I can analyze to look at what duration metadata is stored within the file. The case of YouTube is questionable since I'm not sure whether it is legal for me to download an mp4 from YouTube without going through YouTube's official API.

**Comment by yhcharles** [comment]
Thanks, I can understand your concern.

And further investigation shows it seems not an issue due to just_audio, but may relate to MacOS/iOS. I've tried to play the audio in different apps. When playing in Safari and QuickTime, the duration is wrong, but in Chrome and other third-party players, it is correct. So I guess it's some codec problem from Apple.

BTW, I was able to find a workaround by wrapping the audio in ClippingAudioSource, and set the end time to the correct duration. I'm closing this issue.
